ground/src/main/java/com/google/android/ground/persistence/remote/DataStoreException.kt

```diff
@@ -29,6 +29,10 @@ open class DataStoreException(message: String?) : RuntimeException(message) {
     @JvmStatic
     @Throws(DataStoreException::class)
     fun <T : Any> checkType(expectedType: Class<*>, obj: T): T {
+      // Handle Kotlin Long (java.lang.Long) vs Java primitive long (long)
```

Comment:
Can you please leave a TODO with a bug instead for better code health cleanup?

Reply:
Done